api/src/utils/events.ts

```
@@ -239,10 +241,15 @@ async function listenAndSyncContractEvents() {
       }
     });
   } catch (error) {
+    // If it fails 5 times, exit
```

I'm not sure this is a good idea. The idea behind running this continuously is in case the node is temporarily unavailable or something. I think an exponential backoff before retrying would probably be a good idea, but I would prefer not to hide errors. For example, I'm currently getting this error: `Error while listening for contract events: `TypeError: Cannot read property 'abi' of undefined`, which should be fixed.